gptel-openai.el

```diff
@@ -330,28 +341,49 @@ Mutate state INFO with response metadata."
                                 (point) 'gptel nil (point-min))))
           (pcase (get-char-property (point) 'gptel)
             ('response
-             (push (list :role "assistant"
-                         :content (buffer-substring-no-properties (point) prev-pt))
-                   prompts))
+             (when-let ((content (gptel--trim-prefixes
+                                  (buffer-substring-no-properties (point) prev-pt))))
+               (push (list :role "assistant" :content content) prompts)))
+            (`(tool . ,id)
+             (save-excursion
+               (condition-case-unless-debug _err
+                   (let* ((tool-call (read (current-buffer)))
+                          (id (gptel--openai-format-tool-id id))
+                          (name (plist-get tool-call :name))
+                          (arguments (json-serialize (plist-get tool-call :args)
```

Comment:
If Emacs is built without native json support (either libjansson or the new native parser in Emacs 30) this will fail.  See the macro `gptel--json-encode` near the top of this file used to work around this.